pkg/backup/generative_split_and_scatter_processor.go

```diff
@@ -664,6 +663,10 @@ func runGenerativeSplitAndScatter(
 						if err := chunkEntrySplitAndScatterers[worker].split(ctx, flowCtx.Codec(), splitKey); err != nil {
 							return err
 						}
+
+						if _, err := chunkEntrySplitAndScatterers[worker].scatter(ctx, flowCtx.Codec(), importEntry.Span.Key); err != nil {
+							return err
+						}
 					}
 
 					scatteredEntry := entryNode{
```

Comment:
We might want to verify the effect of this on a larger restore. Namely, it looks to me like we are still going to route this import entry to a worker on the chunk destination which, after this scatter, is less likely to be the leaseholder of the span and I'm not clear on what the impact of that will be on overall restore performance.  I know in your test this didn't seem to impact the overall runtime very much, but I'm not sure that'll hold up when restoring larger backups.